ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto

```diff
@@ -392,19 +465,73 @@ message DamlStateValue {
   }
 }
 
+// These keys identify intents to change the Daml ledger using a Daml
+// transaction built from a DamlCommand.
 message DamlCommandDedupKey {
+  // Intents are namespaced to the list of submitting parties.
   repeated string submitters = 1;
-  reserved 2; // was application_id
+
+  // Submitter provided application identifiers.
+  // Optional. Provided for rank-based deduplication.
+  reserved string application_id = 2;
+
+  // An identifier for the intent.
   string command_id = 3;
 }
 
+// The DamlCommand deduplication logic ensures that Ledger API clients can
+// implement exactly-once submission of DamlCommands using retries.
+// The logic, called rank-based deduplication, works on a
+// per-DamlCommandDedupKey basis. It stores the rank and result of the last
+// submission that was validated, and cancels any further submissions with the
+// same rank instead of executing them.
+//
+// DamlCommandDedupValues are tagged with an expiry date to allow ledgers to
+// reclaim storage space by deleting them. The expiry date is provided either
+// directly or indirectly by Ledger API clients. They are expected to provide
```

Comment:
The expiry date is communicated only with the old ledger API. With the new proposal, the expiry date comes from the (P)KV WriteSerice implementation, doesn't it?

Reply:
True. Will correct.